src/viam/sdk/module/service.cpp

```diff
@@ -249,6 +249,7 @@ void ModuleService_::add_model_from_registry(std::shared_ptr<Server> server, API
     if (services.find(api) == services.end()) {
         add_api_from_registry(server, api);
     }
+    const std::lock_guard<std::mutex> lock(lock_);
```

Comment:
(put lock below the call to `add_api_from_registry` since `add_api_from_registry` locks as well).

Reply:
Good question. I'm not too worried right now, since everything is synchronous and this is taking the lock immediately after `add_api_from_registry` returns. At a minimum this `lock` makes race conditions less likely at least. Might be worth thinking about long term though!

Reply:
Potential bonus points if you make the `inlock` versions take a `const std::lock_guard<mutex>&` as a proof of work, but I don't find it that useful because it doesn't ensure that you have actually taken the right lock.